core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -123,6 +123,16 @@ public static Object fromReader(Reader reader) throws IOException {
             throw new JsonParseException("document doesn't start with a valid json element : "
                     + initialToken, jp.getCurrentLocation());
         }
+        
+        JsonToken t ;
+        try { t = jp.nextToken(); }
```

Comment:
Ahh, I was not taking into account that the illegal case would respond differently to the legal case. 

That also explains why it seems like Jackson should not be continuing at all, depending on how the spec is interpreted, and should terminate regular parsing after the first top level element is fully parsed and throw an exception if any further non-whitespace tokens are found.